modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/raft/snapshot/metrics/RaftSnapshotsMetricsSource.java

```diff
@@ -0,0 +1,297 @@
+     * Decrements the {@code IncomingSnapshotsLoadingMeta} counter.
+     */
+    public void onLoadSnapshotPhaseEnd() {
+        snapshotsLoadingMetaCounter.decrementAndGet();
+    }
+
+    /**
+     * Marks the beginning of the phase where the node waits for catalog to be ready/apply updates
+     * for the incoming snapshot installation.
+     * Increments the {@code IncomingSnapshotsWaitingCatalog} counter.
+     */
+    public void onWaitingCatalogPhaseStart() {
+        snapshotsWaitingCatalogCounter.incrementAndGet();
+    }
+
+    /**
+     * Marks the end of the "waiting for catalog" phase during incoming snapshot installation.
+     * Decrements the {@code IncomingSnapshotsWaitingCatalog} counter.
+     */
+    public void onWaitingCatalogPhaseEnd() {
+        snapshotsWaitingCatalogCounter.decrementAndGet();
+    }
+
+    /**
+     * Marks the beginning of the phase where MV (multi-version) data is loaded from the snapshot.
+     * Increments the {@code IncomingSnapshotsLoadingMvData} counter.
+     */
+    public void onLoadMvDataPhaseStart() {
+        snapshotsLoadingMvDataCounter.incrementAndGet();
+    }
+
+    /**
+     * Marks the end of the phase where MV (multi-version) data is loaded from the snapshot.
+     * Decrements the {@code IncomingSnapshotsLoadingMvData} counter.
+     */
+    public void onLoadMvDataPhaseEnd() {
+        snapshotsLoadingMvDataCounter.decrementAndGet();
+    }
+
+    /**
+     * Marks the beginning of the phase where transaction metadata is loaded from the snapshot.
+     * Increments the {@code IncomingSnapshotsLoadingTxMeta} counter.
+     */
+    public void onLoadTxMetasPhaseStart() {
+        snapshotsLoadingTxMetaCounter.incrementAndGet();
+    }
+
+    /**
+     * Marks the end of the phase where transaction metadata is loaded from the snapshot.
+     * Decrements the {@code IncomingSnapshotsLoadingTxMeta} counter.
+     */
+    public void onLoadTxMetasPhaseEnd() {
+        snapshotsLoadingTxMetaCounter.decrementAndGet();
+    }
+
+    /**
+     * Marks the beginning of preparing indexes for build (for example, collecting row IDs) as part of
+     * incoming snapshot installation.
+     * Increments the {@code IncomingSnapshotsPreparingIndexForBuild} counter.
+     */
+    public void onSetRowIdToBuildPhaseStart() {
+        snapshotsPreparingIndexForBuildCounter.incrementAndGet();
+    }
+
+    /**
+     * Marks the end of preparing indexes for build as part of incoming snapshot installation.
+     * Decrements the {@code IncomingSnapshotsPreparingIndexForBuild} counter.
+     */
+    public void onSetRowIdToBuildPhaseEnd() {
+        snapshotsPreparingIndexForBuildCounter.decrementAndGet();
+    }
+
+    /**
+     * Marks the beginning of the storage preparation phase for incoming snapshot installation.
+     * Increments the {@code IncomingSnapshotsPreparingStorages} counter.
+     */
+    public void onPreparingStoragePhaseStart() {
```

Comment:
Let's move these methods higher. The idea is that order of methods (and fields) reflect (where possible) the order of events happening during a snapshot installation.